### PR TITLE
Fix accessibility issues (part 2)

### LINKED ISSUE #1184: [Deque Analysis] Login module's "serious" accessibility issues
> Serious: This issue results in serious barriers for people with disabilities, and will partially prevent them from accessing fundamental features or content. People relying on assistive technologies will experience significant frustration as a result. Issues falling under this category are major problems, and remediation should be a priority. 

## Deque Analysis Summary

A total of 3 serious issues exist (not including color contrast which are handled in other tickets) in our login module (available in the upper right on each page).

1. ~~(ID: 469328) "Label is not persistent" on the email/password fields. "placeholder is being used as the only visual label for a text field. The placeholder label is not visually available after data is entered in the text field."~~
    * ~~Recommended Fix: "Labels MUST be visible. Ensure that the label is programmatically associated with the form fields via the label/for technique or ARIA attributes."~~
    * FIXED ALREADY VIA Aria attributes (verified as of 7.5, though uncertain when it was actually fixed)
2. (ID 469335) "Identify Input Purpose Not Accurate." on the email/password fields. "Autocomplete attribute to identify the purpose of this form element is not accurate." (Currently both have `autocomplete="off"`)
    * Recommended fix: Fix this issue by using the autocomplete attribute with the appropriate value from HTML5.2 on the text input element. The list of 53 defined input purposes for UI components can be found here -
https://www.w3.org/TR/WCAG21/#input-purposes  (For our purposes, they likely should be set to "username" and "current-password")
3. (ID 469343) ~~"Text spacing content cut off." on the "Have you forgotten your password?" link.~~
    * ~~Recommended fix: Fix this issue by allowing users to override text spacing styles by providing flexibility in the height and width of content containers. Some common techniques include:~~
        * ~~Size containers to a have a value greater than the default width of the text. Small areas of text (such as navigation bars) need to have sufficient space within each box so that words and characters do not expand out of their containers.~~
        * ~~Use the inline-block value of the CSS display property and do not set negative margins to help allow for spacing overrides.~~
        * ~~Do not set the CSS height property, it can help ensure paragraphs are able to expand.~~
        * ~~See also https://www.w3.org/WAI/WCAG21/Understanding/text-spacing~~
        * NO LONGER REPRODUCIBLE IN 7.6.1.  Unsure when fixed.
4. (ID 469229) ~~"id attribute value must be unique". Document has multiple static elements with the same id attribute: `loginDropdownMenu`~~
    * ~~NOTE: Rated a "minor" issue, not "serious", but it may be simple to fix. If it's not simple, it should be delayed & moved to a different ticket.~~
    * FIXED BY #2683

Full list of issues is viewable at (requires login): https://axeauditor.dequecloud.com/test-run/0856438a-a19a-11eb-bc31-b7d5be387c86/issues?sortField=ordinal&sortDir=asc&filter%5Bseverity%5D=3&filter%5Btype%5D=issue&filter%5Bpage_number%5D=8

### LINKED ISSUE #2697: [Accessibility] Empty (invisible) buttons appear on `/community-list`
**Describe the bug**
Originally reported in https://github.com/DSpace/dspace-angular/issues/2055#issuecomment-1402464187

If you use the WAVE browser plugin on the `/community-list` page, it will report several "Empty button" errors.  NOTE: These errors will only be reported for Communities which have displayed "short description" text under the Community name.

These are invisible buttons which have no text or `aria-label`.  They are not seen as errors by all accessibility tools though (e.g. they are not detected by the Deque Axe DevTools browser plugin).

**To Reproduce**
1. Reproducible on the demo site at https://demo.dspace.org/community-list  (or with any local site using our demo entities data set)
2. Enable the WAVE browser plugin on the page.  You will see a number of "Empty button" Errors reported.  (The exact number of errors depends on the number of communities displayed)
    * Click on the "Details" and view only those "Empty Button" errors (deselect everything else)
    * You'll see the errors come from any Community in the list which has a "short description" displayed.

**Expected behavior**
No errors should occur.  It's likely these buttons simply need to have an `aria-label` or some similar descriptive text, so that they are no longer seen as empty.  Alternatively, these buttons could be removed from the page (as they are invisible) if doing so does not impact page behavior.

**Related work**
Related to #2055, where it was originally reported.

### LINKED ISSUE #2775: Accesibility issue in edit/access-control
**Describe the bug**
In the edit section for access control there are two accessibility errors regarding empty labels.

**To Reproduce**
1. Log in as an admin
2. edit any access-control of a community
3. Open WAVE Evaluation Tool

image

**Expected behavior**
There should be no error

@tdonohue i can take this activity

### PR DESCRIPTION
**Pull request by alexandrevryghem**
## References
* Fixes #2697
* Fixes #2775 
* Resolves part of #1176 (469380, 469634)
* Fixes #1184 (469335) -- This fixes the final remaining issue of that ticket, so it can now be closed.
* Related #2683

## Description
This is a follow up PR on #2683, last time I only used Axe DevTools & Ligthouse. This time I used Wave to find additional errors. I didn't fix the header errors and warnings and also didn't fix the contrast issues because seperate PRs lready exists for those.

## Instructions for Reviewers
Instead of making a big list of changes here I seperated the fixes by page/type of issue and used the commit message to describe the fixes to facilitate the review process.

## Checklist
- [x] My PR is small in size (e.g. less than 1,000 lines of code, not including comments & specs/tests), or I have provided reasons as to why that's not possible.
- [x] My PR passes [ESLint](https://eslint.org/) validation using `yarn lint`
- [x] My PR doesn't introduce circular dependencies (verified via `yarn check-circ-deps`)
- [x] My PR includes [TypeDoc](https://typedoc.org/) comments for _all new (or modified) public methods and classes_. It also includes TypeDoc for large or complex private methods.
- [x] My PR passes all specs/tests and includes new/updated specs or tests based on the [Code Testing Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Testing+Guide).
- [x] If my PR includes new libraries/dependencies (in `package.json`), I've made sure their licenses align with the [DSpace BSD License](https://github.com/DSpace/DSpace/blob/main/LICENSE) based on the [Licensing of Contributions](https://wiki.lyrasis.org/display/DSPACE/Code+Contribution+Guidelines#CodeContributionGuidelines-LicensingofContributions) documentation.
- [x] If my PR includes new features or configurations, I've provided basic technical documentation in the PR itself.
- [x] If my PR fixes an issue ticket, I've [linked them together](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue).

### CODE REVIEW

File: src/app/info/feedback/feedback-form/feedback-form.component.html [hunk 1]
```diff
@@ -3,7 +3,7 @@
     <form class="primary" [formGroup]="feedbackForm" (ngSubmit)="createFeedback()">
       <h1>{{ 'info.feedback.head' | translate }}</h1>
       <p>{{ 'info.feedback.info' | translate }}</p>
-      <fieldset class="col p-0">
+      <form class="col p-0">
```
Comment (tdonohue) [inline src/app/info/feedback/feedback-form/feedback-form.component.html:6]: I cannot get the feedback form the work anymore?  It appears this `<form>` is nested under another `<form>` on line 3.  Maybe that's messing things up?

If I visit the feedback form, it loads.  But the "Send Feedback" button will not enable, even if you enter in an email and comments.

**Review by tdonohue — CHANGES_REQUESTED** [review]
@alexandrevryghem : Overall, this looks **great**!  I've verified each commit individually & verified the fixes listed in the description,..and I've only noticed a few minor issues:

1. The feedback form no longer works.  See inline comment below
2. The fix for #1176 seems to be *partial*?  I see the "Results Per Page" in the menu now has an `aria-selected=true` next to the checked value.  However, the "Sort Options" section of that same menu does NOT have that next to the checked value.  So, I think we just need to update the "Sort Options" in the same way.

Everything else looks great.  So once these minor issues can be fixed, I think this can be merged & backported to 7.6.x.

**Comment by alexandrevryghem** [comment]
@tdonohue: Thnx for the feedback! I fixed both issues and amended the fixes for a cleaner history (I only updated the `FeedbackComponent` & the `PaginationComponent`)

**Review by tdonohue — APPROVED** [review]
👍 Thank you @alexandrevryghem !  This looks good now and everything is working as expected.  Merging & auto-porting back to 7.6.x

**Pull request merged**